lib/sparkql/function_resolver.rb

```diff
@@ -795,14 +681,6 @@ def range(start_str, end_str)
     }
   end
 
-  def cast_field(value, type)
-    {
-      :type => :function,
-      :value => "cast",
-      :args => [value, type]
-    }
-  end
```

Comment:
I removed the `*_field` functions here since they are exclusively evaluated on the server.

Reply:
If I understand correctly, your changes build up the function data automatically for the parer's result array? If so, that seems to be a big win!